src/librustdoc/fold.rs

```diff
@@ -51,53 +51,45 @@ pub trait DocFolder: Sized {
                 TraitItem(i)
             }
             ImplItem(mut i) => {
-                if let Ok(re) = Regex::new(r"([^a-z_])Self::([a-zA-Z_])") {
-                    // This whole if block allows to use `Self::<method_name>` in method's doc
-                    // by replacing
-                    // "Self::<method_name>" inside documentation
-                    // with
-                    // <inner.for_.path.segements>::<method>
-                    if let Type::ResolvedPath { path, .. } = &i.for_ {
-                        if let Some(segment) = path.segments.first() {
-                            let replacer = format!("${{1}}{}::${{2}}", segment.name);
-                            i.items = i
-                                .items
+                let matcher_opt = Regex::new(r"([^a-z_])Self::([a-zA-Z_])");
```

Comment:
Use `lazy_static!` as https://docs.rs/regex/1.3.7/regex/#example-avoid-compiling-the-same-regex-in-a-loop points out

Reply:
Good point, thanks!

Reply:
and done